scripts/pre_commit_linter.py

```diff
@@ -520,25 +520,57 @@ def _lint_py_files(config_pylint, config_pycodestyle, files_to_lint, result):
 
         # These lines invoke Pycodestyle.
         style_guide = pycodestyle.StyleGuide(config_file=config_pycodestyle)
-        pycodestyle_report = style_guide.check_files(
-            paths=current_files_to_lint)
-        # This line prints Pycodestyle's output to the console.
-        pycodestyle_report.print_statistics()
+        pycodestyle_report, pycodestyle_results = _check_files(
+            style_guide, paths=current_files_to_lint)
 
         if pylinter.msg_status != 0 or pycodestyle_report.get_count() != 0:
             are_there_errors = True
 
         current_batch_start_index = current_batch_end_index
 
     if are_there_errors:
+        print ''
+        print 'Error Summary:'
+        for pycodestyle_result in pycodestyle_results:
+            print pycodestyle_result
         result.put('%s    Python linting failed' % _MESSAGE_TYPE_FAILED)
     else:
         result.put('%s   %s Python files linted (%.1f secs)' % (
             _MESSAGE_TYPE_SUCCESS, num_py_files, time.time() - start_time))
 
+    print ''
     print 'Python linting finished.'
 
 
+def _check_files(style_guide, paths=None):
+    """Run all pycodestyle checks on the paths."""
+    if paths is None:
+        paths = style_guide.paths
+    report = style_guide.options.report
+    runner = style_guide.runner
+    report.start()
+    all_file_results = []
+    try:
+        for filepath in paths:
+            if os.path.isdir(filepath):
+                style_guide.input_dir(filepath)
+            elif not style_guide.excluded(filepath):
+                runner(filepath)
+                deferred_print = report._deferred_print # pylint: disable=protected-access
+                deferred_print.sort()
```

Comment:
Two questions:

Why is sorting needed?

Is there a way to get deferred_print via a public method/function?

Reply:
I looked into the original implementation [here](https://github.com/PyCQA/pycodestyle/blob/master/pycodestyle.py#L2203). The results are directly printed and not returned/stored. `_deferred_print` was not exposed via any public method or function. I had to re-implement this part so that we can store the results and process it later.
Regarding sorting, I am not quite sure but the original implementation follows a similar way. Therefore, I sorted it here.

Reply:
Better not to reimplement a third-party file (and even if you do, you should always mention in comments that you've done that) -- it can result in tricky situations if the original reference code ends up changing in the future.

I suggest taking a look at swapping out the print statement, i.e. running the function in question using a context manager. You could perhaps store the result to a string or array and then use that variable later. See e.g. this SO link:

https://stackoverflow.com/questions/14197009/how-can-i-redirect-print-output-of-a-function-in-python